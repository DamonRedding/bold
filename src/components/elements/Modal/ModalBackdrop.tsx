import * as React from 'react'

import { withStyles, WithStylesProps } from '../../../styles'

export interface ModalBackdropProps extends WithStylesProps {
    onClick(): any
}

@withStyles
export class ModalBackdrop extends React.PureComponent<ModalBackdropProps> {

    render() {
        const { css, onClick } = this.props
        const styles = {
            backdrop: {
                position: 'fixed',
                zIndex: 1030,
                top: 0,
                left: 0,
                bottom: 0,
                right: 0,
                backgroundColor: 'rgba(0, 0, 0, 0.08)',
            },
        }
        return (
            <div className={css(styles.backdrop)} onClick={onClick} />
        )
    }
}
