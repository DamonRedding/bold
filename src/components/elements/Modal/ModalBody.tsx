import * as React from 'react'

import { withStyles, WithStylesProps } from '../../../styles'

export interface ModalBodyProps extends WithStylesProps {
}

@withStyles
export class ModalBody extends React.PureComponent<ModalBodyProps> {

    render() {
        const { css } = this.props
        const styles = {
            body: {
                padding: '3rem',
            },
        }
        return (
            <div className={css(styles.body)}>
                {this.props.children}
            </div>
        )
    }
}
