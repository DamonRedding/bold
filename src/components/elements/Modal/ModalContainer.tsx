import React from 'react'

import { ExternalStyles, Theme, useStyles } from '../../../styles'
import { Omit } from '../../../util'
import { Button } from '../Button'
import { Icon } from '../Icon'
import { Tooltip } from '../Tooltip'

export interface ModalContainerProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'style'> {
  style?: ExternalStyles
  hasCloseIcon?: boolean
  onClose?(): any
}

export function ModalContainer(props: ModalContainerProps) {
  const { style, onClose, hasCloseIcon, children, ...rest } = props
  const { classes, css } = useStyles(styles)

  return (
    <div role='dialog' aria-modal='true' className={css(classes.wrapper, style)} {...rest}>
      {hasCloseIcon && (
        <Tooltip text='Fechar'>
          <Button aria-label='Fechar' size='small' skin='ghost' style={classes.closeButton} onClick={onClose}>
            <Icon icon='timesDefault' />
          </Button>
        </Tooltip>
      )}

      {children}
    </div>
  )
}

ModalContainer.defaultProps = {
  hasCloseIcon: true,
  onClose: () => null,
} as Partial<ModalContainerProps>

export const styles = (theme: Theme) => ({
  wrapper: {
    border: `1px solid ${theme.pallete.divider}`,
    boxShadow: theme.shadows.outer['160'],
    borderRadius: theme.radius.modal,
    backgroundColor: theme.pallete.surface.main,
    minWidth: 520,
    pointerEvents: 'auto',
  } as React.CSSProperties,
  closeButton: {
    float: 'right',
    marginTop: '0.5rem',
    marginRight: '0.5rem',
  } as React.CSSProperties,
})
