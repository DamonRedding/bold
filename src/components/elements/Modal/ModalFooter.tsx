import React from 'react'

import { withStyles, WithStylesProps } from '../../../styles'

export interface ModalFooterProps extends WithStylesProps {
}

@withStyles
export class ModalFooter extends React.PureComponent<ModalFooterProps> {

    render() {
        const { css, theme } = this.props
        const styles = {
            footer: {
                backgroundColor: theme.pallete.surface.background,
                padding: '1rem',
                borderBottomLeftRadius: theme.radius.modal,
                borderBottomRightRadius: theme.radius.modal,
            },
        }
        return (
            <div className={css(styles.footer)}>
                {this.props.children}
            </div>
        )
    }
}
