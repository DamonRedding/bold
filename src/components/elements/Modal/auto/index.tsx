import * as React from 'react'

import { Omit } from '../../../../util/types'

import { ModalAuto, ModalAutoProps } from './ModalAuto'
import { ModalAutoMountingTarget, ModalAutoRenderProps } from './ModalAutoMountingTarget'

export { ModalAutoMountingTarget, ModalAutoRenderFunction, ModalAutoRenderProps } from './ModalAutoMountingTarget'

export interface ModalConfig extends Omit<ModalAutoProps, 'dispose'> {
}

export const modal = (config: ModalConfig) => {
    return () => {
        const append = ModalAutoMountingTarget.append

        if (!append) {
            throw new Error(`Your must include a <ModalAutoMountingTarget /> component in your application.`)
        }

        append((props: ModalAutoRenderProps) => (
            <ModalAuto
                {...config}
                dispose={props.dispose}
            />
        ))
    }
}
