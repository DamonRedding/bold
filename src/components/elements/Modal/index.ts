export { Modal, ModalProps, ModalSize } from './Modal'
export { ModalBackdrop, ModalBackdropProps } from './ModalBackdrop'
export { ModalBody, ModalBodyProps } from './ModalBody'
export { ModalContainer, ModalContainerProps } from './ModalContainer'
export { ModalControlled, ModalControlledProps, ModalController } from './ModalControlled'
export { ModalFooter, ModalFooterProps } from './ModalFooter'
export * from './auto'
