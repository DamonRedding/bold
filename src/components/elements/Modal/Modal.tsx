import * as React from 'react'

import { withStyles, WithStylesProps } from '../../../styles'

import { ModalBackdrop } from './ModalBackdrop'
import { ModalBody } from './ModalBody'
import { ModalContainer, ModalContainerProps } from './ModalContainer'
import { ModalFooter } from './ModalFooter'

export interface ModalProps extends WithStylesProps, ModalContainerProps {
    open: boolean
    renderFooter?(): React.ReactNode
    onBackdropClick?(): any
}

@withStyles
export class Modal extends React.PureComponent<ModalProps> {

    static defaultProps = {
        onBackdropClick: () => null,
    }

    render() {
        const { open, renderFooter, onBackdropClick, css, theme, ...rest } = this.props
        const styles = {
            wrapper: {
                transition: 'all .2s',
            },
            open: {
                visibility: 'visible',
                opacity: 1,
            },
            close: {
                visibility: 'hidden',
                opacity: 0,
            },
            container: {
                zIndex: theme.zIndex.modalContainer,
                position: 'fixed',
                left: '50%',
                top: '50%',
                transform: 'translate(-50%, -50%)',
            },
        }

        return (
            <div className={css(styles.wrapper, open ? styles.open : styles.close)}>
                <ModalContainer styles={styles.container} {...rest}>
                    <ModalBody>
                        {this.props.children}
                    </ModalBody>

                    {renderFooter &&
                        <ModalFooter>
                            {renderFooter()}
                        </ModalFooter>
                    }
                </ModalContainer>

                <ModalBackdrop onClick={onBackdropClick} />
            </div>
        )
    }
}
