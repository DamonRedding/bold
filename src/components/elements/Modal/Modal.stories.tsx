import { action } from '@storybook/addon-actions'
import { boolean, select, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'

import { withPropTypes, withTheme } from '../../../stories-addons'
import { HFlow } from '../../layout/Flow/HFlow'
import { Button } from '../button/Button/Button'

import { modal } from './auto'
import { ModalAutoMountingTarget } from './auto/ModalAutoMountingTarget'
import { Modal } from './Modal'
import { ModalBody } from './ModalBody'
import { ModalContainer, ModalSize } from './ModalContainer'
import { ModalControlled, ModalController } from './ModalControlled'
import { ModalFooter } from './ModalFooter'

const sizes: { [key in ModalSize]: any } = {
    'large': 'large',
    'small': 'small',
}

let controller1: ModalController = null
let controller2: ModalController = null

const createFooter = (ctrl: ModalController = null) => (
    <HFlow justifyContent='flex-end'>
        <Button label='Cancel' onClick={ctrl && ctrl.close || action('cancel clicked')} />
        <Button label='Save' type='primary' onClick={ctrl && ctrl.close || action('save clicked')} />
    </HFlow>
)

storiesOf('Components/Modal', module)
    .addDecorator(withPropTypes())
    .addDecorator(withKnobs)
    .addDecorator(withTheme())
    .add('default', () => (
        // tslint:disable jsx-no-lambda
        <Modal
            open={boolean('open', true)}
            size={select('size', sizes, 'large')}
            renderFooter={createFooter}
        >
            Lorem ipsum.
        </Modal>
    ))
    .add('auto', () => (
        <>
            <Button
                label='Auto modal'
                onClick={modal({
                    size: 'small',
                    render: () => 'Confirm?',
                    actions: [
                        { label: 'Cancel', onClick: action('Cancel') },
                        { label: 'Ok', type: 'primary', onClick: action('Ok') },
                    ],
                })}
            />
            <ModalAutoMountingTarget />
        </>
    ))
    .add('controlled/trigger', () => (
        // tslint:disable jsx-no-lambda
        <HFlow>
            <Button label='Modal with actions' onClick={() => controller1.open()} />
            <Button label='Simple Modal' onClick={() => controller2.open()} />

            <ModalControlled
                control={(ctrl) => controller1 = ctrl}
                size={select('size', sizes, 'large')}
                renderFooter={createFooter}
            >
                Lorem ipsum.
            </ModalControlled>

            <ModalControlled
                control={(ctrl) => controller2 = ctrl}
                size={select('size', sizes, 'large')}
            >
                Lorem ipsum.
            </ModalControlled>
        </HFlow>
    ))
    .add('parts', () => (
        <ModalContainer size={select('size', sizes, 'large')} onClose={action('onClose')}>
            <ModalBody>
                Teste
            </ModalBody>
            <ModalFooter>
                <HFlow justifyContent='flex-end'>
                    <Button label='Secondary' />
                    <Button label='Primary' type='primary' />
                </HFlow>
            </ModalFooter>
        </ModalContainer>
    ))
