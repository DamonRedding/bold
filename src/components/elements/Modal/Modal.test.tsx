import { mount, ReactWrapper, render, shallow } from 'enzyme'
import * as React from 'react'

import { withTheme } from '../../../test'
import { Button } from '../button/Button/Button'

import { Modal } from './Modal'
import { ModalBody } from './ModalBody'
import { ModalContainer } from './ModalContainer'
import { ModalFooter } from './ModalFooter'

describe('Modal', () => {

    const createFooter = () => (
        <Button label='Action' />
    )

    it('should render closed', () => {
        expect(render(withTheme(
            <Modal open={false} renderFooter={createFooter}>
                Testing.
            </Modal>
        ))).toMatchSnapshot()
    })

    it('should render open', () => {
        expect(render(withTheme(
            <Modal open={true} renderFooter={createFooter}>
                Testing.
            </Modal>
        ))).toMatchSnapshot()
    })

    it('should use modal parts', () => {
        const wrapper = mount(withTheme(
            <Modal open={true} renderFooter={createFooter}>
                Testing.
            </Modal>
        ))
        expect(wrapper.find(ModalContainer).length).toEqual(1)
        expect(wrapper.find(ModalBody).length).toEqual(1)
        expect(wrapper.find(ModalFooter).length).toEqual(1)
    })

    it('should not render footer if renderFooter is not provided', () => {
        const noFooter = mount(withTheme(
            <Modal open={true}>
                Testing.
            </Modal>
        ))
        expect(noFooter.find(ModalFooter).length).toEqual(0)
    })
})
