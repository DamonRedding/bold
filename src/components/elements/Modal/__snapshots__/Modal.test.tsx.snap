// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render closed 1`] = `
.emotion-8 {
  -webkit-transition: all .2s;
  transition: all .2s;
  visibility: hidden;
  opacity: 0;
}

.emotion-6 {
  border: 1px solid #e6e6e6;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  border-radius: 2px;
  background-color: #ffffff;
  position: relative;
  width: 768px;
  z-index: 1040;
  position: fixed;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.emotion-2 {
  background-color: transparent;
  border: none;
  border-radius: 4px;
  color: #666666;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  position: absolute;
  top: 1rem;
  right: 1rem;
}

.emotion-2 span {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: color .2s;
  transition: color .2s;
}

.emotion-2:disabled {
  opacity: 0.5;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0 0 0 2px #056DFF;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-3 {
  padding: 3rem;
}

.emotion-5 {
  background-color: #f2f2f7;
  padding: 1rem;
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
}

.emotion-4 {
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: "Noto Sans",sans-serif;
  line-height: 1.5rem;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  background-color: #ffffff;
  border: 1px solid #b3b3b3;
  border-radius: 4px;
  color: #666666;
  font-size: 0.875rem;
  padding: calc(0.75rem - 1px) 2.5rem;
}

.emotion-4:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-4:not(:disabled) {
  cursor: pointer;
}

.emotion-4 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: color .2s;
  transition: color .2s;
}

.emotion-4 > span > :not(:last-child) {
  margin-right: 0.5rem;
}

.emotion-4:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0 0 0 2px #f2f2f7,0 0 0 4px #056DFF;
}

.emotion-4:not(:disabled):hover {
  background-color: #ebebeb;
}

.emotion-7 {
  position: fixed;
  z-index: 1030;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-color: rgba(0,0,0,0.08);
}

<div
  class="emotion-8"
>
  <div
    class="emotion-6"
  >
    <button
      class="emotion-2"
      type="button"
    >
      <span>
        <span
          class="emotion-1"
        >
          <svg
            aria-hidden="true"
            class="emotion-0"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M17.71 7.71l-1.42-1.42-4.29 4.3-4.29-4.3-1.42 1.42 4.3 4.29-4.3 4.29 1.42 1.42 4.29-4.3 4.29 4.3 1.42-1.42-4.3-4.29 4.3-4.29z"
              id="Fechar"
            />
          </svg>
        </span>
      </span>
    </button>
    <div
      class="emotion-3"
    >
      Testing.
    </div>
    <div
      class="emotion-5"
    >
      <button
        class="emotion-4"
        type="button"
      >
        <span>
          <span>
            Action
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    class="emotion-7"
  />
</div>
`;

exports[`Modal should render open 1`] = `
.emotion-6 {
  border: 1px solid #e6e6e6;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  border-radius: 2px;
  background-color: #ffffff;
  position: relative;
  width: 768px;
  z-index: 1040;
  position: fixed;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.emotion-2 {
  background-color: transparent;
  border: none;
  border-radius: 4px;
  color: #666666;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  position: absolute;
  top: 1rem;
  right: 1rem;
}

.emotion-2 span {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: color .2s;
  transition: color .2s;
}

.emotion-2:disabled {
  opacity: 0.5;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0 0 0 2px #056DFF;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-3 {
  padding: 3rem;
}

.emotion-5 {
  background-color: #f2f2f7;
  padding: 1rem;
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
}

.emotion-4 {
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: "Noto Sans",sans-serif;
  line-height: 1.5rem;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  background-color: #ffffff;
  border: 1px solid #b3b3b3;
  border-radius: 4px;
  color: #666666;
  font-size: 0.875rem;
  padding: calc(0.75rem - 1px) 2.5rem;
}

.emotion-4:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-4:not(:disabled) {
  cursor: pointer;
}

.emotion-4 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: color .2s;
  transition: color .2s;
}

.emotion-4 > span > :not(:last-child) {
  margin-right: 0.5rem;
}

.emotion-4:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0 0 0 2px #f2f2f7,0 0 0 4px #056DFF;
}

.emotion-4:not(:disabled):hover {
  background-color: #ebebeb;
}

.emotion-7 {
  position: fixed;
  z-index: 1030;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-color: rgba(0,0,0,0.08);
}

.emotion-8 {
  -webkit-transition: all .2s;
  transition: all .2s;
  visibility: visible;
  opacity: 1;
}

<div
  class="emotion-8"
>
  <div
    class="emotion-6"
  >
    <button
      class="emotion-2"
      type="button"
    >
      <span>
        <span
          class="emotion-1"
        >
          <svg
            aria-hidden="true"
            class="emotion-0"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M17.71 7.71l-1.42-1.42-4.29 4.3-4.29-4.3-1.42 1.42 4.3 4.29-4.3 4.29 1.42 1.42 4.29-4.3 4.29 4.3 1.42-1.42-4.3-4.29 4.3-4.29z"
              id="Fechar"
            />
          </svg>
        </span>
      </span>
    </button>
    <div
      class="emotion-3"
    >
      Testing.
    </div>
    <div
      class="emotion-5"
    >
      <button
        class="emotion-4"
        type="button"
      >
        <span>
          <span>
            Action
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    class="emotion-7"
  />
</div>
`;
